### PR TITLE
chore: drop dependency on playwright-core in all our packages

### LINKED ISSUE #2268: [BUG] Types don't work in WebStorm ("Unresolved function or method")
**Context:**
- Playwright Version: 1.0.2
- IDE: WebStorm 2020.1

This isn't really a bug in Playwright, but the TypeScript types aren't picked up in WebStorm because they're in a different package (`playwright-core`).

image

Although it's not the 'fault' of Playwright, I've never seen any other package do this, so Playwright will be one of the few packages where types don't work for WebStorm users.

I've logged this as an issue with JetBrains: https://youtrack.jetbrains.com/issue/WEB-45571, but it's not considered a bug so isn't going to change.

For WebStorm users finding this, the workaround is to either add `playwright-core` to your `package.json` or mark `node_modules/playwright-core` as 'Not excluded' in the IDE:
image

Then all is well:
image

If this could be resolved that'd be great, but otherwise maybe note this in the getting started?

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch:
- drops dependency on playwright-core in all our packages. Instead of
  the dependency, packages are now built with `//packages/build_package.sh`
  script.
- unifies `browsers.json` - now there's a single `//browsers.json` file
  that is used to manage browser revisions.

This patch temporary switches canary publishing to `--dryn-run` from CI/CD so that we
can verify that it does sane things.

We'll unify all our package management scripts under `//packages/` in a
follow-up.

Fixes #2268

### CODE REVIEW

File: packages/build_package.sh [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+if [[ "${PACKAGE_NAME}" == "playwright" ]]; then
+  PACKAGE_DESCRIPTION="A high-level API to automate web browsers"
+  PACKAGE_BROWSERS="['chromium', 'firefox', 'webkit']"
+
+  # For Playwright, we need to copy README.md
+  README_PATH="${PACKAGE_PATH}/README.md"
+  CLEANUP_PATHS+=("${README_PATH}")
+  cp ../README.md "${README_PATH}"
+elif [[ "${PACKAGE_NAME}" == "playwright-core" ]]; then
+  PACKAGE_DESCRIPTION="A high-level API to automate web browsers"
```
Comment (yury-s) [inline packages/build_package.sh:79]: Should we put a note that this package should not be used directly?

**Review by yury-s — APPROVED** [review]

File: packages/build_package.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+else
+  echo "ERROR: unknown package '${PACKAGE_NAME}'"
+  exit 1
+fi
+
+
+# 4. Build root package unless --no-build was provided.
+
+if [[ -z "${NO_BUILD}" ]]; then
+  npm run build
```
Comment (aslushnikov) [inline packages/build_package.sh:99]: 1. types are missing
2. .npmignore is missing
3. do not `npm run clean` and do not `npm run build`

File: packages/build_package.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  // 8. Move result to the outputPath
+  fs.renameSync(path.join(packagePath, tgzName), outputPath);
+  console.log(outputPath);
+})();
+
+async function writeToPackage(fileName, content) {
+  const toPath = path.join(packagePath, fileName);
+  cleanupPaths.push(toPath);
+  console.error(`- generating: //${path.relative(ROOT_PATH, toPath)}`);
+  await new Promise((resolve, reject) => {
```
Comment (dgozman) [inline packages/build_package.js:139]: Why not use promisify?

File: packages/build_package.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    });
+  });
+}
+
+async function copyToPackage(fileOrDirectoryName) {
+  const fromPath = path.join(ROOT_PATH, fileOrDirectoryName);
+  const toPath = path.join(packagePath, fileOrDirectoryName);
+  cleanupPaths.push(toPath);
+  console.error(`- copying: //${path.relative(ROOT_PATH, fromPath)} -> //${path.relative(ROOT_PATH, toPath)}`);
+  await new Promise((resolve, reject) => {
```
Comment (dgozman) [inline packages/build_package.js:154]: Same here.

File: utils/publish_all_packages.sh [hunk 5]
```diff
@@ -66,11 +75,16 @@ else
   exit 1
 fi
 
-npm run clean
-npm publish .                            --tag="${NPM_PUBLISH_TAG}"
-npm publish packages/playwright-firefox  --tag="${NPM_PUBLISH_TAG}"
-npm publish packages/playwright-webkit   --tag="${NPM_PUBLISH_TAG}"
-npm publish packages/playwright-chromium --tag="${NPM_PUBLISH_TAG}"
-npm publish packages/playwright          --tag="${NPM_PUBLISH_TAG}"
+PLAYWRIGHT_TGZ="$(node ./packages/build_package.js playwright ./playwright.tgz)"
```
Comment (dgozman) [inline utils/publish_all_packages.sh:78]: Do we need to update the recipe on the bot to build beforehand?

**Review by dgozman — APPROVED** [review]

File: .npmignore [hunk 6]
```diff
@@ -12,6 +12,10 @@ lib/injected/
 
 # root for "playwright-core" package
 !index.js
+!install.js
+!README.md
```
[the remark this answers is not in the document]
Comment (aslushnikov) [reply .npmignore:16]: : D whitelisting doesn't hurt, I guess :)

File: .npmignore [hunk 7]
```diff
@@ -12,6 +12,10 @@ lib/injected/
 
 # root for "playwright-core" package
```
[the remark this answers is not in the document]
Comment (aslushnikov) [reply .npmignore:13]: Done.

**Pull request merged**